filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -192,6 +192,41 @@ function createArticleNode($doc, $pubObject) {
 			$this->appendTextMiningCollectionNodes($doc, $articleNode, $article, $submissionGalleys);
 		}
 
+		$issueId = $article->getCurrentPublication()->getData('issueId');
+		if ($cache->isCached('issues', $issueId)) {
+		    $issue = $cache->get('issues', $issueId);
+		} else {
+		    $issueDao = DAORegistry::getDAO('IssueDAO'); /* @var $issueDao IssueDAO */
+		    $issue = $issueDao->getById($issueId, $context->getId());
+		    if ($issue) $cache->add($issue, null);
+		}
+		$journalId = $issue->getData('journalId');
+		if(!empty($journalId)){
+		    $accessRights = null;
+		    $journalDao = DAORegistry::getDAO('JournalDAO'); /** @var JournalDAO $journalDao */
+		    $journal = $journalDao->getById($journalId);
+		    if($journal->getData('publishingMode') == 0){
+		        $accessRights = 'openAccess';
+		    } else if($journal->getData('publishingMode') == 1) {
+		        if ($issue->getAccessStatus() == 1) {
+		            $accessRights = 'openAccess';
+		        } else if ($issue->getAccessStatus() == 2) {
+		            if ($article->getCurrentPublication()->getData('accessStatus') == 1) {
+		                $accessRights = 'openAccess';
+		            }
+		        }
+		    }
+		    if($accessRights == 'openAccess' || !empty($journal->getData('licenseUrl'))){
```

Comment:
You should also check the publication licenseUrl -- it could override the journal settings:
$rightsURL = $article->getCurrentPublication()->getData('licenseUrl') ?? $context->getData('licenseUrl');